osmoutils/accum/accum_helpers.go

```diff
@@ -12,7 +12,7 @@ var (
 
 // Creates a new position or override an existing position
 // at accumulator's current value with a specific number of shares and unclaimed rewards
-func initOrUpdatePosition(accum AccumulatorObject, accumulatorValue sdk.DecCoins, index string, numShareUnits sdk.Dec, unclaimedRewards sdk.DecCoins, options *Options) {
+func InitOrUpdatePosition(accum AccumulatorObject, accumulatorValue sdk.DecCoins, index string, numShareUnits sdk.Dec, unclaimedRewards sdk.DecCoins, options *Options) {
```

Comment:
I would like to suggest that we make a copy of this method in `genesis.go` in CL.

It is important that we keep this unexported. Breakages of this abstraction lead to misuse and nasty bugs that are difficult to find and debug

Reply:
Agreed, unexported this and added a func directly to genesis.go https://github.com/osmosis-labs/osmosis/pull/5024/commits/c7d6a52cd9f5fb578338141f8269ce014602d86b